module-apps/application-call/ApplicationCall.cpp

```diff
@@ -205,6 +205,12 @@ namespace app
         return ret;
     }
 
+    sys::MessagePointer ApplicationCall::handleAppClose(sys::Message *msgl)
+    {
+        callModel->hangUpCall();
```

Comment:
does this check in call machine if the call is happening before sending AT+CHUP? (just asking out of curiosity :P )

Reply:
No it doesn't do any check. It cause send AT+CHUP in any state of call machine besides Idle.